src/superstream_producer.rs

```diff
@@ -0,0 +1,135 @@
+use crate::error::ProducerCloseError;
+use crate::{
+    client::Client,
+    environment::Environment,
+    error::{ProducerCreateError, ProducerPublishError},
+    producer::{ConfirmationStatus, NoDedup, Producer},
+    superstream::{DefaultSuperStreamMetadata, RoutingStrategy},
+};
+use rabbitmq_stream_protocol::message::Message;
+use std::collections::HashMap;
+use std::future::Future;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+//type FilterValueExtractor = Arc<dyn Fn(&Message) -> String + 'static + Send + Sync>;
+
+#[derive(Clone)]
+pub struct SuperStreamProducer<T>(
+    Arc<SuperStreamProducerInternal>,
+    HashMap<String, Producer<T>>,
+    DefaultSuperStreamMetadata,
+    PhantomData<T>,
+);
+
+/// Builder for [`SuperStreamProducer`]
+pub struct SuperStreamProducerBuilder<T> {
+    pub(crate) environment: Environment,
+    //pub filter_value_extractor: Option<FilterValueExtractor>,
+    pub routing_strategy: RoutingStrategy,
+    pub(crate) data: PhantomData<T>,
+}
+
+pub struct SuperStreamProducerInternal {
+    pub(crate) environment: Environment,
+    client: Client,
+    // TODO: implement filtering for superstream
+    //filter_value_extractor: Option<FilterValueExtractor>,
+    routing_strategy: RoutingStrategy,
+}
+
+impl SuperStreamProducer<NoDedup> {
```

Comment:
@wolf4ood I can see that Producer is implemented also with type Dedup, but there is no function in environment to create it as Dedup actually. I'm wondering if this is really necessary or we just leave NoDedup for the moment

Reply:
Let's just keep it Nodedup for now